### PR TITLE
Debug mode, Test failure annotations, Code Coverage

### PR DESCRIPTION
**Pull request by displague**
* Adds:
  * Debug mode (https://www.terraform.io/docs/extend/debugging.html#enabling-debugging-in-a-provider)
  * ~Test failure annotations (https://github.com/marketplace/actions/golang-test-annotations) #171~
  * Code Coverage (https://app.codecov.io/gh/equinix/terraform-provider-metal)
* Increases test duration to 180m due to recent timeouts.

### CODE REVIEW

**Comment by displague** [comment]
One drawback of `-json` and outputting the results directly to a file is that we can't monitor the progress of tests.  However, monitoring the progress of tests is not a great experience today because GHA doesn't reveal the output lines that were missed before loading the page until the job completes.

If we were to pipe to `tee`, we could see the output but in the `-json` format, they may be less readable.

File: .github/workflows/acctest.yml [hunk 1]
```diff
@@ -68,10 +68,25 @@ jobs:
 
         METAL_AUTH_TOKEN: ${{ secrets.PACKET_AUTH_TOKEN }}
       run: |
-        go test -v -cover -parallel 4 -timeout 120m ./metal
+        go test -v -json -coverprofile coverage.txt -covermode=atomic -parallel 4 -timeout 180m ./metal > test.json
```
Comment (displague) [inline .github/workflows/acctest.yml:71]: Do coverage.txt and test.json have to be unique names for the TF 0.13 test and 1.0?

Comment (displague) [reply .github/workflows/acctest.yml:71]: Can we increase the parallelization?

File: .github/workflows/acctest.yml [hunk 2]
```diff
@@ -68,10 +68,25 @@ jobs:
...
+        test-results: test.json
     - name: Sweeper
       if: ${{ always() }}
       env:
         METAL_AUTH_TOKEN: ${{ secrets.PACKET_AUTH_TOKEN }}
       run: |
-        go test ./metal -v -sweep="tf_test"
+        go test ./metal -v -json -sweep="tf_test"  > sweep.json
+    - name: Annotate sweep
+      if: always()
```
Comment (displague) [inline .github/workflows/acctest.yml:84]: This is a different format than the existing `${{ always() }}`.  I assume they are both acceptable.

File: .github/workflows/acctest.yml [hunk 3]
```diff
@@ -68,10 +68,25 @@ jobs:
...
+      uses: guyarb/golang-test-annotations@v0.4.0
+      with:
+        test-results: sweep.json
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v2
+      with:
+        token: ${{ secrets.CODECOV_TOKEN }}
```
Comment (displague) [inline .github/workflows/acctest.yml:91]: This token has been added to the repo settings.

**Comment by displague** [comment]
Hmm, none of the errors were annotated.

**Comment by displague** [comment]
> Hmm, none of the errors were annotated.

Annotation is only possible when the Test itself was changed.  The code to annotate must appear on the PR.

There are other GH actions available that would provide a summary of the test run and we may want to look into those.

**Review by t0mk — APPROVED** [review]
The debug mode code looks OK, although I havent' really tried to run the provide with delve. I've never used it and I'm not sure how I would go about it, but maybe it's a good alternative to debug prints (whihc are hard to do in a tf provider).

I tried to produce the coverage file and it created it alright. I didn't try to upload it to codecov.

Overall LGTM.

**Comment by displague** [comment]
We should start seeing codecov posting on PRs - might need to activate the [webhook](https://app.codecov.io/gh/equinix/terraform-provider-metal/settings/general) if not.

[codecov](https://codecov.io/gh/equinix/terraform-provider-metal)

**Pull request merged**